### PR TITLE
 Added classifyDifficulty feature

### PR DESCRIPTION
**Pull request by Dhairya-Kanabar**
This pull request introduces the `classifyDifficulty` function to classify Sudoku puzzles based on the number of clues and backtracking steps taken during the solving process.

### Changes:
- Implemented `classifyDifficulty` function.
  - The difficulty is calculated by subtracting the clue count from 40 and adding the integer division of backtracking steps by 10.
  - The difficulty levels are classified as:
    - "Easy" (score < 10)
    - "Medium" (10 ≤ score < 20)
    - "Hard" (20 ≤ score < 30)
    - "Expert" (score ≥ 30)

### Test Cases:
- Added sample test cases to validate that the function correctly classifies puzzles with varying clue counts and backtracking steps.

This feature can now be integrated into the Sudoku solving process to dynamically determine puzzle difficulty.

### CODE REVIEW

**Comment by soni-shashan** [comment]
### @Dhairya-Kanabar mention the issue in comment always something like `closes #issue-number` or `fixes #issue-number`, which will automatically close the issue ex: `fixes #40 : Added classifyDifficulty feature`

**Comment by soni-shashan** [comment]
### @Dhairya-Kanabar , you added Python code to this project, but the Sudoku project contains only C. While your logic is correct, please implement this feature in C within the `sudoku.c` file.

File: .vscode/c_cpp_properties.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      "cStandard": "${default}",
+      "cppStandard": "${default}",
+      "intelliSenseMode": "windows-gcc-x64",
+      "compilerArgs": [
+        ""
+      ]
+    }
+  ],
+  "version": 4
+}
```
Comment (shreylakhtaria) [inline .vscode/c_cpp_properties.json:18]: Add Last empty Line.

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      "setupCommands": [
+        {
+          "description": "Enable pretty-printing for gdb",
+          "text": "-enable-pretty-printing",
+          "ignoreFailures": true
+        }
+      ]
+    }
+  ]
+}
```
Comment (shreylakhtaria) [inline .vscode/launch.json:24]: Add Last empty Line.

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    "**/.vscode",
+    "**/.vscode/**"
+  ],
+  "C_Cpp_Runner.useAddressSanitizer": false,
+  "C_Cpp_Runner.useUndefinedSanitizer": false,
+  "C_Cpp_Runner.useLeakSanitizer": false,
+  "C_Cpp_Runner.showCompilationTime": false,
+  "C_Cpp_Runner.useLinkTimeOptimization": false,
+  "C_Cpp_Runner.msvcSecureNoWarnings": false
+}
```
Comment (shreylakhtaria) [inline .vscode/settings.json:59]: Add Last empty Line.

File: sudoku.c [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#include <stdio.h>
+
 #include "sudoku.h"
 
 int isValid(Sudoku *s, int row, int col, int num) {
```
Comment (shreylakhtaria) [inline sudoku.c:4]: Use Bollean Insted of number and Also add return Type.

**Review by shreylakhtaria — CHANGES_REQUESTED** [review]

File: sudoku_classification.py
Comment (soni-shashan) [inline sudoku_classification.py:1]: ### You need to write a function in sudoku.c

File: .vscode/c_cpp_properties.json
Comment (soni-shashan) [inline .vscode/c_cpp_properties.json:1]: ### add in `.vscode` in the `.gitignore` file to clean project structure

File: .vscode/launch.json
Comment (soni-shashan) [inline .vscode/launch.json:1]: ### add in `.vscode` in the `.gitignore` file to clean project structure

File: .vscode/settings.json
Comment (soni-shashan) [inline .vscode/settings.json:1]: ### add in `.vscode` in the `.gitignore` file to clean project structure

File: sudoku.c
Comment (soni-shashan) [inline sudoku.c:1]: ### call `sudoku_classification` function in `main.c` after print grid

**Review by soni-shashan — CHANGES_REQUESTED** [review]
### kindly check changes

**Pull request closed without merging**